ApeWorX/ape
# feat: support contract namespaces

### What I did

* Allows interface-based projects to use namespace-techniques to avoid duplicate contract names. This allows the re-use of `interface.json` so long as it is in a different directory.

### How I did it

* Create a bridge class `ContractNamespace` that serves up the next node (either another `ContractNamespace` or the `ContractContainer` (the leaf).
* Change `pm` first-class plugin to honor set the full `.`-based path contract names.

### How to verify it

Create a project like this:

```yaml
contracts:
  namespace0:
    - inteface.json
  namespace1:
     namespace:2
        - interface.json
```

See test for fuller example!

**Also, this will help with cairo specifically**:  https://github.com/ApeWorX/ape-cairo/pull/6

### Checklist
